### PR TITLE
point users to required_multi_asset_neighbors in error

### PR DESCRIPTION
**Pull request by benpankow**
## Summary

Adds a call-out to use `required_multi_asset_neighbors` if selecting a subset of non-subsettable multi-assets.

## Test Plan

Tested locally

Screenshot 2024-03-04 at 2 33 20 PM

### CODE REVIEW

**Comment by benpankow** [comment]
* **#20245** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/20245?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @benpankow and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: python_modules/dagster/dagster/_core/definitions/asset_layer.py [hunk 1]
```diff
@@ -971,7 +971,8 @@ def _subset_assets_defs(
                 f"{sorted(list(asset.check_keys))}, but "
                 f"attempted to select only {sorted(list(selected_subset))}. "
                 "This AssetsDefinition does not support subsetting. Please select all "
-                "asset keys produced by this asset."
+                "asset keys produced by this asset.\n\nIf using an AssetSelection, you may "
+                "append required_multi_asset_neighbors() to select any remaining assets, for example:\nAssetSelection.groups('my_group').upstream().required_multi_asset_neighbors()"
```
Comment (sryza) [inline python_modules/dagster/dagster/_core/definitions/asset_layer.py:975]: Wrap plz

Comment (sryza) [inline python_modules/dagster/dagster/_core/definitions/asset_layer.py:975]:
```suggestion
                "use required_multi_asset_neighbors() to select any remaining assets, for example:\nAssetSelection.keys('my_asset').required_multi_asset_neighbors()"
```

**Review by sryza — CHANGES_REQUESTED** [review]
A few small suggestions

**Review by sryza — APPROVED** [review]

**Pull request merged**